src/hooks/swap/useSyncController.ts

```diff
@@ -22,10 +22,10 @@ export default function useSyncController({ value, settings }: SwapController):
   }, [settings, value])
 
   const setSwap = useUpdateAtom(swapAtom)
-  setSwap(() => value)
+  if (value) setSwap(() => value)
 
   const setSettings = useUpdateAtom(settingsAtom)
-  setSettings(() => settings)
+  if (settings) setSettings(() => settings)
```

Comment:
these read like it might be impossible to clear state

Reply:
you're 100% correct - I've reverted this file in this PR.